### ISSUE TITLE
Missing image in an old glowfic throws uncaught exception

### ISSUE THREAD

**Issue by drowdemon** [rocurley/glowfic-dl#1]
At least, the title is my guess at the diagnoses. I wrapped the relevant lines in a try catch that just caught all exceptions because I don't actually know the libraries this depends on. This allowed the script to complete, albeit very slowly as it tried for a long time to download images that didn't exist before giving up.

Thanks again for creating this script, it's very helpful!

My ad-hoc change is summarized as follows:

```
for (k, v) in tqdm(image_map.map.items()):
        try:
            resp = requests.get(k)
            item = epub.EpubItem(
                uid=v,
                file_name=v,
                media_type=resp.headers["Content-Type"],
                content=resp.content,
            )
            resp.close()
            book.add_item(item)
        #except TimeoutError: # this didn't work
        except:
            pass
```

Error output was as follows

> Traceback (most recent call last):
>   File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 169, in _new_conn
>     conn = connection.create_connection(
>   File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 96, in create_connection
>     raise err
>   File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 86, in create_connection
>     sock.connect(sa)
> TimeoutError: [Errno 110] Connection timed out
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 699, in urlopen
>     httplib_response = self._make_request(
>   File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 382, in _make_request
>     self._validate_conn(conn)
>   File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 1012, in _validate_conn
>     conn.connect()
>   File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 353, in connect
>     conn = self._new_conn()
>   File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 181, in _new_conn
>     raise NewConnectionError(
> urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f5a973fde20>: Failed to establish a new connection: [Errno 110] Connection timed out
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/lib/python3/dist-packages/requests/adapters.py", line 439, in send
>     resp = conn.urlopen(
>   File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 755, in urlopen
>     retries = retries.increment(
>   File "/usr/lib/python3/dist-packages/urllib3/util/retry.py", line 574, in increment
>     raise MaxRetryError(_pool, url, error or ResponseError(cause))
> urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='hollow-art.com', port=443): Max retries exceeded with url: /files/bases/2017/09/20170923/26332/aimee-garcia-lucifer-3404570.jpg (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f5a973fde20>: Failed to establish a new connection: [Errno 110] Connection timed out'))
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/hdd/Code/glowfic-dl/main.py", line 181, in <module>
>     main()
>   File "/hdd/Code/glowfic-dl/main.py", line 155, in main
>     resp = requests.get(k)
>   File "/usr/lib/python3/dist-packages/requests/api.py", line 76, in get
>     return request('get', url, params=params, **kwargs)
>   File "/usr/lib/python3/dist-packages/requests/api.py", line 61, in request
>     return session.request(method=method, url=url, **kwargs)
>   File "/usr/lib/python3/dist-packages/requests/sessions.py", line 542, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/usr/lib/python3/dist-packages/requests/sessions.py", line 655, in send
>     r = adapter.send(request, **kwargs)
>   File "/usr/lib/python3/dist-packages/requests/adapters.py", line 516, in send
>     raise ConnectionError(e, request=request)
> requests.exceptions.ConnectionError: HTTPSConnectionPool(host='hollow-art.com', port=443): Max retries exceeded with url: /files/bases/2017/09/20170923/26332/aimee-garcia-lucifer-3404570.jpg (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f5a973fde20>: Failed to establish a new connection: [Errno 110] Connection timed out'))

**Comment by rocurley** [comment]
Iiinteresting. Can you send me the glowfic in question so I can reproduce this? Email is in my profile if you'd prefer not to post it here.

**Comment by ApriiSR** [comment]
mad investor chaos has this problem

**Comment by rocurley** [comment]
Should be fixed in e3782a21ccfb3d7df90e839cef972c2ed86a122c.
